src/Cool/DidYouKnow/didYouKnowModal.cpp

```diff
@@ -0,0 +1,98 @@
+#include "didYouKnowModal.hpp"
+#include <imgui.h>
+#include <vcruntime.h>
+#include <string>
+
+void DidYouKnowModal::open()
+{
+    auto difference = std::chrono::system_clock::to_time_t(std::chrono::system_clock::now()) - std::chrono::system_clock::to_time_t(_current_timestamp);
+
+    size_t time_to_wait = 1 * 60 * 60; // 1 hour
+
+    if (!_has_been_opened && difference > time_to_wait)
+    {
+        _has_been_opened = true;
+        ImGui::OpenPopup(_id.c_str());
+        _current_timestamp = std::chrono::system_clock::now();
+        return;
+    }
+}
+
+void DidYouKnowModal::prepare_next_tip()
+{
+    _current_tip_index++;
+    if (_current_tip_index >= _all_tips.size())
+        _current_tip_index = 0;
+    _text = _all_tips[_current_tip_index];
```

Comment:
You don't need to copy the text here, not store it in the class. You can just get it thanks to the index when necessary.
Maybe you can make a method `current_tip()`.